specification/keyvault/data-plane/Security.KeyVault.SecurityDomain/client.tsp

```diff
@@ -11,19 +11,28 @@ using KeyVault;
 @TypeSpec.Versioning.useDependency(KeyVault.Versions.`v7.5`)
 namespace ClientCustomizations;
 
+@@access(CertificateInfoObject, Access.public);
+@@access(SecurityDomainObject, Access.public);
+
 @client({
-  name: "SecurityDomainClient",
+  name: "KeyVaultClient",
```

Comment:
Why? This should be renamed. We don't want other languages generating `KeyVaultClient`.

Reply:
We have to patch the generated client to apply our challenge auth policy, provide functioning pollers (because of `download` having an unusual return type, as well as the "Success" terminal condition being non-standard), and fix operation return types (again because of `download`): https://github.com/Azure/azure-sdk-for-python/blob/56b3fd9d9ae75a6fc05c82b5caf5a2eb1f403ac1/sdk/keyvault/azure-keyvault-securitydomain/azure/keyvault/securitydomain/_patch.py

The patched client is the one that users see, and type resolution gets confused if the client we're patching has the same name. I just reused the "KeyVaultClient" name for consistency with generated clients in other Python libraries, but the public client is `SecurityDomainClient`.

Reply:
We could make this client modification language-specific in the future if other languages start generating this library.

Reply:
Can you just use the `@clientName` decorator for Python now? I worry down the road no one is going to remember the context. Seems you're the only one who needs a specialized name, so why make every other language opt into renaming it (since there's no language exclusion)?